restartmanager/restartmanager.go

```diff
@@ -70,10 +71,16 @@ func (rm *restartManager) ShouldRestart(exitCode uint32, hasBeenManuallyStopped
 	if executionDuration.Seconds() >= 10 {
 		rm.timeout = 0
 	}
-	if rm.timeout == 0 {
+	switch {
+	case rm.timeout == 0:
 		rm.timeout = defaultTimeout
-	} else {
+	case rm.timeout < maxRestartTimeout:
 		rm.timeout *= backoffMultiplier
+		if rm.timeout > maxRestartTimeout {
+			rm.timeout = maxRestartTimeout
+		}
+	case rm.timeout > maxRestartTimeout:
```

Comment:
nit: How can this ever be called? If you want to be safe then maybe remove this case and move `if rm.timeout > maxRestartTimeout` on line 79 outside of the switch.